lib/adf_info.py

```diff
@@ -117,16 +117,17 @@ def __init__(self, config_file, debug=False):
         #Case names:
         case_names = self.get_cam_info('cam_case_name', required=True)
 
-        print(case_names)
-
         #Grab test case nickname(s)
-        test_nicknames = self.get_cam_info('case_nickname')
-        if test_nicknames is None:
-            test_nicknames = [] #Re-set to be an empty list
-            for case_name in case_names:
-                test_nicknames.append(case_name)
-            #End for
-        #End if
+        test_nickname_list = self.get_cam_info('case_nickname')
+
+        test_nicknames = [] #set to be an empty list
+        for i,nickname in enumerate(test_nickname_list):
+            if nickname is None:
+                test_nicknames.append(case_names[i])
+            else:
+                test_nicknames.append(test_nickname_list[i])
+            #End if
+        #End for
```

Comment:
I realized that we'll still want to check if `test_nicknames_list` is None in case the `case_nickname` config variable isn't present at all:

```suggestion
        if test_nickname_list:
            test_nicknames = [] #set to be an empty list
            for i,nickname in enumerate(test_nickname_list):
                if nickname is None:
                    test_nicknames.append(case_names[i])
                else:
                    test_nicknames.append(test_nickname_list[i])
                #End if
            #End for
        else:
            test_nicknames = [] #Re-set to be an empty list
            for case_name in case_names:
                test_nicknames.append(case_name)
            #End for
        #End if
```

Reply:
Ah, good call. I forgot that check, thanks!